File: src/main/java/com/github/weewar/mapviewer/service/WeewarMapRenderer.java
package com.github.weewar.mapviewer.service;

import com.github.weewar.mapviewer.exceptions.ImageNotFoundException;
import com.github.weewar.mapviewer.exceptions.ImageResizeException;
import com.github.weewar.mapviewer.model.WeewarMap;

import java.awt.image.BufferedImage;

public interface WeewarMapRenderer {
    BufferedImage render(WeewarMap weewarMap) throws ImageNotFoundException;

    BufferedImage renderThumbnail(WeewarMap weewarMap, int width, int height) throws ImageNotFoundException, ImageResizeException;
}

File: src/main/kotlin/com/github/weewar/mapviewer/service/WeewarMapRenderer.kt
package com.github.weewar.mapviewer.service

import com.github.weewar.mapviewer.model.*
import java.awt.image.BufferedImage

interface WeewarMapRenderer {
    @Throws(ImageNotFoundException::class)
    fun render(weewarMap: WeewarMap): BufferedImage

    @Throws(ImageNotFoundException::class, ImageResizeException::class)
    fun renderThumbnail(weewarMap: WeewarMap, width: Int, height: Int): BufferedImage
}
